index.js

```diff
@@ -212,26 +205,22 @@ NYC.prototype.instrumentAllFiles = function (input, output, cb) {
 }
 
 NYC.prototype.walkAllFiles = function (dir, visitor) {
-  this.exclude.globSync(dir).forEach(filename => {
-    visitor(filename)
+  this.exclude.globSync(dir).forEach(relFile => {
+    visitor(relFile)
   })
 }
 
-NYC.prototype._maybeInstrumentSource = function (code, filename, relFile) {
-  var instrument = this.exclude.shouldInstrument(filename, relFile)
-  if (!instrument) {
-    return null
-  }
+NYC.prototype._transform = function (code, filename) {
+  const extname = path.extname(filename).toLowerCase()
+  const transform = this.transforms[extname] || (() => null)
 
-  var ext, transform
-  for (ext in this.transforms) {
-    if (filename.toLowerCase().substr(-ext.length) === ext) {
-      transform = this.transforms[ext]
-      break
-    }
-  }
+  return transform(code, { filename: filename })
+}
+
+NYC.prototype._maybeInstrumentSource = function (code, filename) {
+  const instrument = this.exclude.shouldInstrument(filename)
```

Comment:
I think it would be simpler / more readable to just do:
```js
if (!this.exclude.shouldInstrument(filename)) {
  return null;
}
```

That way we avoid the ternary statement.